src/components/SQForm/__tests__/SQFormDropdown.test.js

```diff
@@ -0,0 +1,202 @@
+});
+
+it('should display an empty option when displayEmpty is true', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    displayEmpty: true
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList.children.length).toBe(4);
+  expect(optionsList.childNodes[0]).toHaveTextContent('- -');
+});
+
+it('should not be selectable if it is disabled', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    isDisabled: true
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  expect(expandButton).toHaveAttribute('aria-disabled', 'true');
+});
+
+it('should be selectable if it is not disabled', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    isDisabled: false
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  expect(expandButton).not.toHaveAttribute('aria-disabled', 'true');
+});
+
+it('should display icon and text if field is required', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    isRequired: true
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const required = screen.getByText(/required/i);
+  expect(required).toBeVisible();
+});
+
+it('should not display icon and text if field is not required', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    isRequired: false
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const required = screen.queryByText(/required/i);
+  expect(required).toBeNull();
+});
+
+it('should highlight field if required but no value selected', () => {
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    isRequired: true,
+    validationSchema: {
+      dropdownExample: Yup.string().required('Required')
+    }
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+
+  userEvent.tab();
+  expect(expandButton).toHaveFocus();
+
+  userEvent.tab();
+  expect(expandButton).not.toHaveFocus();
+
+  const required = screen.getByText(/required/i);
+  expect(required).toBeVisible();
+  //expect required to be orange / highlighted
+  //expect(required).toHaveAttribute('class', 'MuiFormHelperText-root Mui-error Mui-required');
```

Comment:
nit: I suspect these comments can be removed

Reply:
or perhaps they are the part of what you would like to be testing?

Reply:
Yes, this is the last piece I think I need to get a whole test. I haven't been able to find a working solution to check that the field is in an error state / highlighted. The only thing I could see that changed was adding `.Mui-error`.

Reply:
Yeah, that's how I tested it: `.toHaveClass(Mui-error)`

Reply:
So what's going wrong is that the `Mui-error` class isn't being applied. So the test fails because it's not there. So I'm somehow not triggering the error state. But I wrote it how I made it happen testing manually. So that's where I'm stuck.

Reply:
Oh, I think it may be because `validationSchema` should be passed to SQForm instead of the SQFormDropdown